src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -58,29 +62,70 @@ export class FilterTimestamp extends HTMLElement {
     );
   }
 
-  onClearFilter = () => {
-    this.startTimeLabel.textContent = 'From Beginning';
-    this.stopTimeLabel.textContent = 'Up to End';
-    this.filterContainer.hidden = true;
+  onClearFilter(clearStart, clearStop) {
+    if (clearStart) {
+      this.startTimeLabel.textContent = 'From Beginning';
+      this.clearStartTimeBtn.hidden = true;
+      delete this.timeStamp.start;
+    }
+
+    if (clearStop) {
+      this.stopTimeLabel.textContent = 'Up to End';
+      this.clearStopTimeBtn.hidden = true;
+      delete this.timeStamp.stop;
+    }
+
+    if (!clearStart && !clearStop) {
+      this.startTimeLabel.textContent = 'From Beginning';
+      this.stopTimeLabel.textContent = 'Up to End';
+      this.timeStamp = {};
+    }
```

Comment:
@atiqueahmedziad at a first glance I would say that this shouldn't be needed, in which case we would need this along with the other two at lines 66-69 and lines 72-76?

Reply:
Yes, Thanks for the idea. :-)